tests/unit/downloader/test_data_collector.py

```diff
@@ -0,0 +1,60 @@
+import json
+import unittest
+from unittest.mock import MagicMock
+
+from ingest.api.ingestapi import IngestApi
+from ingest.downloader.data_collector import DataCollector
+
+
+class DataCollectorTest(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.maxDiff = None
+        self.mock_ingest_api = MagicMock(spec=IngestApi)
+        self.data_collector = DataCollector(self.mock_ingest_api)
+
+    def test_collected_project_and_biomaterials_data_by_submission_uuid_returns_correctly(self):
+        #given
+        project_uuid = '1234'
+        with open('../resources/mock_submission.json') as file:
```

Comment:
I would extract these data preparation bits into meaningfully named function, so this function has the appropriate level of abstraction rather than too many details.

Reply:
This is a test code. I think it is all right like this. If it would be an implementation code, then I would do that.

Reply:
OK, I moved it

Reply:
Also, this class contains only a short test case. I would consider doing it if it would have more than 1 case. Doing it now would be a premature optimization/refactor.

Reply:
Test code does not mean it should be messy.
Tests that are written in an unclear way are difficult to maintain and fix.
You, as the tests, writer, is not the sole audience of the test, but it is other team members in the present and future.
Properly engineering the tests is a standard of established development processes and experienced professional teams.

Reply:
We can leave this for now.